proposed/standard-ci-env-variables.md

```diff
@@ -0,0 +1,47 @@
+# Standardized Environment Variables for CI Services
+
+Many CI services set environment variables that can be used by developer builds. Many of the environment variables are common in scope and theme across CI services, however, the specific variable names are different. We need a set of standardized environment variables across CI services so that an application platform like .NET can rely on them, and provide a common user experience across CI services.
+
+## Context
+
+The .NET Core SDK is implementing new scenarios that require source control manager information. An initial idea was to call out to specific source control tools to get this information. This approach is problematic since it requires an implementation for each source control manager and it has the potential to be fragile and slow.
+
+It turns out that CI services all of the information that these new .NET Core scenarios need via environment variables. Also, there is a strong relationship between these scenarios and official CI-provided builds (much less on local developer builds). As a result, relying on the CI-provided environment variables is attractive.
+
+To make it possible to provide application platform provided experiences across CI services, we need a standardized set of environment variables that are supported across those same CI services. Ideally, this set of environment variables would be supported across multiple CI services and useful for multiple application environments, not just .NET.
+
+## Proposed environment variables
+
+The .NET Core SDK needs are oriented around source control. As a result, the intial list is source control oriented, but there is no affinity to source control on the general idea of standardized environment variables.
+
+It is important that these environment variables do not conflict with other variables. To avoid that, all environment variables will be prepended with "STANDARDCI-". This name is a first proposal for the idea and it may get changed based on feedback.
+
+* **STANDARDCI\_REPOSITORYCOMMITID** -- Commit hash / ID; Example: 2ba93796dcf132de447886d4d634414ee8cb069d
+* **STANDARDCI\_REPOSITORYROOT** -- Root of repository; Example: D:\repos\corefx
+* **STANDARDCI\_REPOSITORYNAME** -- Name repository; Example: dotnet\corefx
+* **STANDARDCI\_REPOSITORYURI** -- Uri for repository; Example: https://github.com/dotnet/corefx
+
+## Support from CI Services
+
+This plan will only work if CI services decide to support these environment variables. An important question is whether CI services have similar environment variables already. The table below suggests that the information we need is already available. An arbitrary sample of CI services were picked for this exercise.
```

Comment:
If the CI services don't add support for the new env vars could the .NET SDK could translate these variables below to the `STANDARDCI_` ones and re-export them?

Reply:
@akoeplinger In theory we could, but then we would have variables specific to some set of CI systems in the .NET SDK, which is undesirable.

Reply:
Ya. Doing that is kinda the opposite of what we want from this plan.